src/party_one.rs

```diff
@@ -136,14 +136,36 @@ pub struct EphKeyGenSecondMsg {}
 //****************** End: Party One structs ******************//
 
 impl KeyGenFirstMsg {
-    pub fn create_commitments() -> (KeyGenFirstMsg, CommWitness, EcKeyPair) {
-        let base: GE = ECPoint::generator();
 
-        let secret_share: FE = ECScalar::new_random();
-        //in Lindell's protocol range proof works only for x1<q/3
-        let secret_share: FE =
-            ECScalar::from(&secret_share.to_big_int().div_floor(&BigInt::from(3)));
+    //in Lindell's protocol range proof works only for x1 \in {q/3 , ... , 2q/3}
+    pub fn is_secret_share_in_range(secret_share: &FE) -> bool {
```

Comment:
can you have a look in `sample_range` in curv and adapt your code, seems that is what we want